### DISCUSSION TITLE
Ory Network B2B marketplace authentication use case

### DISCUSSION THREAD

**Discussion by Big-Brother-Man** [ory/.github#58] [Q&A] [answered]
Hi all, I hope you all are doing well.

I got a question about building a market place using Ory cloud.

What is the best way to go about doing it?

Multiple Ory cloud instances i.e. 1 for customers and another one for b2b?

Different Ory schemas? 

Constraint: I would like to build further b2b offering like b2b forum for example (where only b2b users can enter).

Or is this just and authorization issue i.e. keep all authentication in same bucket but keep having distinct roles?

Many thanks!

**Comment by vinckr** [comment, marked as answer]
Hello @Big-Brother-Man 
apologies for the late answer. Have a look at this blogpost: https://www.ory.sh/run-sso-oauth2-enterprise-server-open-source/. That described a setup for B2B multi-tenancy using separate sub-brands. You can also solve it with one Ory Network instance.

**Reply by Big-Brother-Man** [reply]
hey @vinckr sorry for the delay myself. 

I think this solves my problem. But for now I decided to have all in the same bucket with the differentiator being the public meta data (store specific sign up data in their public meta data).

I think I will move to this approach once we reached enough users. Hopefully should be easy enough to migrate existing users to a different ory domain.

**Reply by vinckr** [reply]
Using the metadata to differentiate is another good approach! But at scale it could be good to split it up a bit, that is true.
